### PR TITLE
Always warn about uncommitted changes at the start of dune-release distrib

### LINKED ISSUE #320: distrib does not see `(name <name>)` stanza if not committed
## Setup

We set up a simple repo with two .opam files. We add a name to
dune-project but do not commit it.

~~~
> git clone git@github.com:lehy/dune-release-repro.git
[snip]
> cd dune-release-repro
liba/  libb/  CHANGES  dune-project  dune-workspace  liba.opam  libb.opam
> cat dune-project 
(lang dune 2.7)
~~~

~~~
> echo '(name toto)' >> dune-project
> cat dune-project
(lang dune 2.7)

(name toto)
~~~

`dune-release tag` works:

~~~
> dune-release tag        
[-] Extracting tag from first entry in CHANGES
[-] Using tag "1.0.0"
[?] Create git tag 1.0.0 for HEAD? [Y/n]
y
[+] Tagged HEAD with version 1.0.0
~~~

## Problem

`dune-release distrib` does not see the uncommitted name. It would be
nice if it could complain about it if committing is really necessary,
or at least mention that committing is necessary in the error message.

~~~
> dune-release distrib
[-] Building source archive
Error: The project name is not defined, please add a (name <name>) field to
your dune-project file.
dune-release: [ERROR] run ['dune' 'subst']: exited with 1
~~~

Committing the dune-project file with a name makes dune-release
distrib work (however the name must be the name of one of the .opam
files).

### PR DESCRIPTION
**Pull request by lehy**
This fixes #320. Otherwise `dune-release distrib` silently ignores uncommitted changes to dune-project (like a `(name <name>)` line for instance).

### CODE REVIEW

**Comment by lehy** [comment]
> Could you please either disable the warning in dry-run mode as used to be
> done or fix the other tests so the warning doesn't show up there?

The reason I removed the check of dry-run for the warning is that #320 happens exactly in the same way if you pass --dry-run.

So I propose to rather fix the other tests (by generating a .gitignore that suppresses the warning, which is mostly due to the mdx infrastructure). This is what I did in the above update.

File: tests/bin/distrib-name/run.t
Comment (lehy) [reply tests/bin/distrib-name/run.t]: Done

File: tests/bin/distrib-name/run.t
Comment (lehy) [reply tests/bin/distrib-name/run.t]: Done

File: tests/bin/distrib-name/run.t
Comment (lehy) [reply tests/bin/distrib-name/run.t]: Done

File: CHANGES.md
Comment (gpetiot) [inline CHANGES.md:51]: This should bt #325 (we put the PR id)

**Review by gpetiot — APPROVED** [review]
Thanks, it looks good to me! Maybe wait until #327 is merged and update the tests before merging this one.

File: CHANGES.md
Comment (lehy) [reply CHANGES.md]: Done.

File: CHANGES.md
Comment (gpetiot) [inline CHANGES.md:50]: Oh by the way it should be `dune-release distrib` instead

Comment (lehy) [reply CHANGES.md]: Done (and rebased). Thank you Guillaume and Nathan, and everybody involved in dune-release. It makes my life easier and I can spend more time coding and less time releasing. Joyeux Noël / bonne fêtes !

Comment (gpetiot) [reply CHANGES.md]: Thank you for your contribution and your patience through the review process! Joyeux Noël !

**Pull request merged**
